### ISSUE TITLE
Extend `Option` with `ok_or_eyre`?

### ISSUE THREAD

**Issue by LeoniePhiline** [eyre-rs/eyre#125]
When discussing https://docs.rs/eyre/latest/eyre/index.html#context-and-option, a user remarked that a closure and macro invocation just for a static string error (as documented: `opt.ok_or_else(|| eyre!("new error message"))`) seemed cumbersome.

The reasoning at the documentation's "`Context` and `Option`" section does make sense - `WrapErr` would indicate wrapping an error, where there is (literally) `None`. This issue is _not_ asking to implement `WrapErr` for `Option`.

An extension trait, offering `ok_or_eyre` on the `Option` type does seem useful to avoid closure + macro boilerplate:

```rust
use std::fmt::{Debug, Display};

pub trait OptionExt<T> {
    /// Example usage:
    ///
    /// ```rust
    /// let option: Option<()> = None;
    ///
    /// let result = option.ok_or_eyre("static str error");
    ///
    /// assert_eq!(result.unwrap_err().to_string(), "static str error");
    /// ```
    fn ok_or_eyre<M>(self, message: M) -> crate::Result<T>
    where
        M: Debug + Display + Send + Sync + 'static;
}

impl<T> OptionExt<T> for Option<T> {
    fn ok_or_eyre<M>(self, message: M) -> crate::Result<T>
    where
        M: Debug + Display + Send + Sync + 'static,
    {
        match self {
            Some(ok) => Ok(ok),
            None => Err(crate::Report::msg(message)),
        }
    }
}
```

Is this something you would consider accepting as contribution? Names are of course up for bikeshedding.

**Comment by yaahc** [comment]
yes! I think this is a fantastic way to navigate that concern; I'd definitely not object to this addition.

**Comment by LeoniePhiline** [comment]
Landed with #129's merge.
